examples/game/breakout.rs

```diff
@@ -5,268 +5,381 @@ use bevy::{
             },
-            ..Default::default()
-        },
-        ..Default::default()
-    });
+            collides: Collides,
+        }
+    }
+}
 
-    // Add walls
-    let wall_material = materials.add(Color::rgb(0.8, 0.8, 0.8).into());
-    let wall_thickness = 10.0;
-    let bounds = Vec2::new(900.0, 600.0);
+fn spawn_walls(mut commands: Commands, mut materials: ResMut<Assets<ColorMaterial>>) {
+    let material_handle = materials.add(config::WALL_COLOR.into());
 
-    // left
-    commands
-        .spawn_bundle(SpriteBundle {
-            material: wall_material.clone(),
-            transform: Transform::from_xyz(-bounds.x / 2.0, 0.0, 0.0),
-            sprite: Sprite::new(Vec2::new(wall_thickness, bounds.y + wall_thickness)),
-            ..Default::default()
-        })
-        .insert(Collider::Solid);
-    // right
-    commands
-        .spawn_bundle(SpriteBundle {
-            material: wall_material.clone(),
-            transform: Transform::from_xyz(bounds.x / 2.0, 0.0, 0.0),
-            sprite: Sprite::new(Vec2::new(wall_thickness, bounds.y + wall_thickness)),
-            ..Default::default()
-        })
-        .insert(Collider::Solid);
-    // bottom
-    commands
-        .spawn_bundle(SpriteBundle {
-            material: wall_material.clone(),
-            transform: Transform::from_xyz(0.0, -bounds.y / 2.0, 0.0),
-            sprite: Sprite::new(Vec2::new(bounds.x + wall_thickness, wall_thickness)),
-            ..Default::default()
-        })
-        .insert(Collider::Solid);
-    // top
-    commands
-        .spawn_bundle(SpriteBundle {
-            material: wall_material,
-            transform: Transform::from_xyz(0.0, bounds.y / 2.0, 0.0),
-            sprite: Sprite::new(Vec2::new(bounds.x + wall_thickness, wall_thickness)),
-            ..Default::default()
-        })
-        .insert(Collider::Solid);
-
-    // Add bricks
-    let brick_rows = 4;
-    let brick_columns = 5;
-    let brick_spacing = 20.0;
-    let brick_size = Vec2::new(150.0, 30.0);
-    let bricks_width = brick_columns as f32 * (brick_size.x + brick_spacing) - brick_spacing;
-    // center the bricks and move them up a bit
-    let bricks_offset = Vec3::new(-(bricks_width - brick_size.x) / 2.0, 100.0, 0.0);
-    let brick_material = materials.add(Color::rgb(0.5, 0.5, 1.0).into());
-    for row in 0..brick_rows {
-        let y_position = row as f32 * (brick_size.y + brick_spacing);
-        for column in 0..brick_columns {
+    commands.spawn_bundle(WallBundle::new(Side::Top, &material_handle));
+    commands.spawn_bundle(WallBundle::new(Side::Bottom, &material_handle));
+    commands.spawn_bundle(WallBundle::new(Side::Left, &material_handle));
+    commands.spawn_bundle(WallBundle::new(Side::Right, &material_handle));
+}
+
+fn spawn_bricks(mut commands: Commands, mut materials: ResMut<Assets<ColorMaterial>>) {
+    let brick_material = materials.add(config::BRICK_COLOR.into());
+
+    // Brick layout constants
```

Comment:
Why didn't you put those in config module?

Reply:
I wasn't sure I could get *everything* in there. I agree with you though, and I'll make that change.